src/main/java/org/b612foundation/adam/datamodel/TargetedPropagation.java

```diff
@@ -0,0 +1,130 @@
+   * which is a local orbital coordinate frame that has the x-axis along the velocity vector, W along the orbital
+   * angular momentum vector, and N completes the right handed system.
+   */
+  private PropagationParameters initialPropagationParameters;
+
+  /**
+   * Parameters for the actual targeting.
+   */
+  private TargetingParameters targetingParameters;
+
+  /** Human-readable description of this object. */
+  private String description;
+
+  /**
+   * The resulting ephemeris from the targeter. This will be the nominal ephemeris if the targeter is configured to
+   * propagate only the nominal trajectory, otherwise the targeted ephemeris.
+   */
+  private String ephemeris;
+
+  /**
+   * The calculated maneuver. Currently will only contain a modification from the initial maneuver in the x-direction.
+   */
+  private double maneuverX;
+  private double maneuverY;
+  private double maneuverZ;
+
+  public PropagationParameters getInitialPropagationParameters() {
+    return initialPropagationParameters;
+  }
+
+  public TargetedPropagation setInitialPropagationParameters(PropagationParameters initialPropagationParameters) {
+    this.initialPropagationParameters = initialPropagationParameters;
+    return this;
+  }
+
+  public TargetingParameters getTargetingParameters() {
+    return targetingParameters;
+  }
+
+  public TargetedPropagation setTargetingParameters(TargetingParameters targetingParameters) {
+    this.targetingParameters = targetingParameters;
+    return this;
+  }
+
+  public String getDescription() {
+    return description;
+  }
+
+  public TargetedPropagation setDescription(String description) {
+    this.description = description;
+    return this;
+  }
+
+  public String getEphemeris() {
+    return ephemeris;
+  }
+
+  public TargetedPropagation setEphemeris(String ephemeris) {
+    this.ephemeris = ephemeris;
+    return this;
+  }
+
+  public double getManeuverX() {
+    return maneuverX;
+  }
+
+  public TargetedPropagation setManeuverX(double maneuverX) {
+    this.maneuverX = maneuverX;
+    return this;
+  }
+
+  public double getManeuverY() {
+    return maneuverY;
+  }
+
+  public TargetedPropagation setManeuverY(double maneuverY) {
+    this.maneuverY = maneuverY;
+    return this;
+  }
+
+  public double getManeuverZ() {
+    return maneuverZ;
+  }
+
+  public TargetedPropagation setManeuverZ(double maneuverZ) {
+    this.maneuverZ = maneuverZ;
+    return this;
+  }
+
+  @Override
+  public int hashCode() {
+    return Objects.hash(super.hashCode(), initialPropagationParameters, targetingParameters, description, ephemeris,
+        maneuverX, maneuverY, maneuverZ);
+  }
+
+  @Override
+  public boolean equals(Object obj) {
+    if (this == obj)
+      return true;
+    if (obj == null)
+      return false;
+    if (getClass() != obj.getClass())
+      return false;
+    TargetedPropagation other = (TargetedPropagation) obj;
```

Comment:
I'm curious... do we need to test this cast? That is - does Java take care of it as a not-equals if it's the wrong type?

Reply:
In general it's a good idea to test a cast. But that's what happened just above, with if (getClass() != obj.getClass()). So since we know this is a TargetedPropagation, we know this is a safe cast.